notebook/analysis/benchmark_results.py

```diff
@@ -0,0 +1,591 @@
+plot += geom_boxplot()
+plot += ggtitle('Effect of subsetting healthy data on prediction accuracy')
+print(plot)
+
+# %%
+plot = ggplot(sepsis_metrics, aes(x='factor(healthy_used)', y='accuracy', fill='supervised')) 
+plot += geom_violin()
+plot += ggtitle('Effect of subsetting healthy data on prediction accuracy')
+print(plot)
+
+# %%
+plot = ggplot(sepsis_metrics, aes(x='factor(healthy_used)', y='balanced_accuracy', fill='supervised')) 
+plot += geom_violin()
+plot += ggtitle('Effect of subsetting healthy data on prediction accuracy')
+print(plot)
+
+# %%
+sepsis_stat_df = create_dataset_stat_df(sepsis_metrics, 
+                                        sample_to_study, 
+                                        sample_metadata, 
+                                        sample_to_label,
+                                        'sepsis')
+
+sepsis_stat_df.tail(5)
+
+# %%
+ggplot(sepsis_stat_df, aes(x='train_val_diff', 
+                           y='accuracy', 
+                           color='val_disease_count')) + geom_point() + facet_grid('model ~ .')
+
+# %%
+plot = ggplot(sepsis_metrics, aes(x='train sample count', y='balanced_accuracy', color='supervised')) 
+plot += geom_smooth()
+plot += geom_point(alpha=.2)
+plot
+
+# %% [markdown]
+# ## Same Distribution Tuberculosis
+
+# %%
+in_files = glob.glob('../../results/keep_ratios.tb*')
+print(in_files[:5])
+
+# %%
+tb_metrics = pd.DataFrame()
+for path in in_files:
+    new_df = pd.read_csv(path, sep='\t')
+    model_info = path.strip('.tsv').split('tb.')[-1]
+    model_info = model_info.split('.')
+        
+    supervised_model = model_info[0]
+             
+    new_df['supervised'] = supervised_model
+    
+    new_df['seed'] = model_info[-2]
+        
+    tb_metrics = pd.concat([tb_metrics, new_df])
+    
+tb_metrics = tb_metrics.rename({'fraction of data used': 'healthy_used'}, axis='columns')
+tb_metrics['healthy_used'] = tb_metrics['healthy_used'].round(1)
+tb_metrics
+
+# %%
+plot = ggplot(tb_metrics, aes(x='factor(healthy_used)', y='accuracy')) 
+plot += geom_boxplot()
+plot += ggtitle('Effect of subsetting healthy data on prediction accuracy')
+print(plot)
+
+# %%
+plot = ggplot(tb_metrics, aes(x='factor(healthy_used)', y='accuracy', fill='supervised')) 
+plot += geom_violin()
+plot += ggtitle('Effect of subsetting healthy data on prediction accuracy')
+print(plot)
+
+# %%
+plot = ggplot(tb_metrics, aes(x='factor(healthy_used)', y='balanced_accuracy', fill='supervised')) 
+plot += geom_violin()
+plot += ggtitle('Effect of subsetting healthy data on prediction accuracy')
+print(plot)
+
+# %%
+tb_stat_df = create_dataset_stat_df(tb_metrics, 
+                                    sample_to_study, 
+                                    sample_metadata, 
+                                    sample_to_label,
+                                    'tb')
+
+tb_stat_df.tail(5)
+
+# %%
+ggplot(tb_stat_df, aes(x='train_val_diff', 
```

Comment:
Do you think this plot would change at all with balanced accuracy on the Y axis?

Reply:
I updated the plot, and it looks about like I would expect. The overall trend changes a little bit, as seen in the plot afterwards, but mainly it just gets rid of the accuracy 1 runs where the model guesses the positive class everytime for LR. The deepnet still runs into that, but it's split between accuracy 1 and accuracy .5 depending on which class it decides to guess each time